extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/DevConsoleCORSFilter.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.vertx.http.runtime.cors.CORSConfig;
+import io.quarkus.vertx.http.runtime.cors.CORSFilter;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+
+public class DevConsoleCORSFilter extends CORSFilter {
+
+    private static final String LOCAL_HOST = "localhost";
+    private static final String HTTP_LOCAL_HOST = "http://" + LOCAL_HOST;
+    private static final String HTTPS_LOCAL_HOST = "https://" + LOCAL_HOST;
+
+    public DevConsoleCORSFilter() {
+        super(corsConfig());
+    }
+
+    private static CORSConfig corsConfig() {
+        CORSConfig config = new CORSConfig();
+        config.origins = Optional.of(List.of(HTTP_LOCAL_HOST, HTTPS_LOCAL_HOST));
+        return config;
+    }
+
+    @Override
+    public void handle(RoutingContext event) {
+        HttpServerRequest request = event.request();
+        HttpServerResponse response = event.response();
+        String origin = request.getHeader(HttpHeaders.ORIGIN);
+        if (origin == null) {
+            event.next();
+        } else {
+            if (!origin.contains(LOCAL_HOST)) {
+                response.end();
```

Comment:
You might want to add a 'Only localhost origin is allowed' error message, as devs might be confused why `http://127.0.0.1/q/dev/` no longer works.

Reply:
Or you could do a DNS lookup on the origin and make sure it resoles to loopback.

Reply:
I guess the message should be enough for now, as the browser should not be even producing `Origin` in the usual DevUI interaction with the user... I'd be concerned the DNS lookup can just hang or something like that...

Reply:
does this mean 0.0.0.0:8080/q/dev will stop working too?

Reply:
@maxandersen No, This PR will affect neither `http://127.0.0.1:8080/q/dev/` nor `0.0.0.0:8080/q/dev/` since DevUI pages are not implemented as SPA using XHR or Fetch scripts, so `Origin` will not be even produced by the browser. For the next version of DevUI the Origin check may need to be adjusted.

Reply:
websockets might break, and also things like the config editor as it does use JS. @phillip-kruger will know the details.

Reply:
By websockets I mean the log streaming.

Reply:
I can test this PR for the Web Sockets (log streaming and continues testing use ws) and a few extensions use js post. I'll report back here